selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/dashboard/NewWorkspace.java

```diff
@@ -428,4 +1083,117 @@ public void waitPageLoad(int timeout) {
+    return seleniumWebDriverHelper
+        .waitPresenceOfAllElements(By.xpath("//div[@data-stack-id]"))
+        .stream()
+        .map(webElement -> webElement.getAttribute("data-stack-id"))
+        .collect(toList());
+  }
+
+  public void waitStackSelected(StackId stackId) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> {
+                  return seleniumWebDriverHelper
+                      .waitPresenceOfAllElements(By.xpath("//div[@data-stack-id]"))
+                      .stream()
+                      .filter(
+                          webElement ->
+                              webElement.getAttribute("data-stack-id").equals(stackId.toString()))
+                      .collect(Collectors.toList())
+                      .get(0)
+                      .getAttribute("class")
+                      .contains("stack-selector-item-selected");
+                });
+  }
+
+  public List<String> getVisibleStacks() {
+    return seleniumWebDriverHelper
+        .waitPresenceOfAllElements(By.xpath("//div[@data-stack-id]"))
+        .stream()
+        .filter(seleniumWebDriverHelper::isVisible)
+        .map(webElement -> webElement.getAttribute("data-stack-id"))
+        .collect(Collectors.toList());
+  }
+
+  public void waitVisibleStacks(List<String> expectedVisibleStacks) {
+    webDriverWaitFactory
+        .get()
+        .until(
+            (ExpectedCondition<Boolean>)
+                driver -> getVisibleStacks().equals(expectedVisibleStacks));
+  }
+
+  public void waitStacks(List<String> expectedStacks) {
+    expectedStacks.forEach(
+        stackId ->
+            seleniumWebDriverHelper.waitPresence(
+                By.xpath(format("//div[@data-stack-id='%s']", stackId))));
+  }
+
+  public int getAvailableStacksCount() {
+    return getAvailableStacks().size();
+  }
+
+  public void waitStacksNotPresent(List<String> stacksIdForChecking) {
+    stacksIdForChecking.forEach(
+        stackId ->
+            webDriverWaitFactory
+                .get()
+                .until(
+                    (ExpectedCondition<Boolean>)
+                        driver -> !getAvailableStacks().contains(stackId)));
+  }
+
+  public void waitStacksOrder(List<String> expectedOrder) {
+    webDriverWaitFactory
+        .get()
+        .until((ExpectedCondition<Boolean>) driver -> expectedOrder.equals(getAvailableStacks()));
+  }
+
+  public void clickNameButton() {
+    seleniumWebDriverHelper.waitAndClick(By.xpath("//div[@che-column-title='Name']/div"));
+  }
+
+  public void clickOnTitlePlaceCoordinate() {
+    seleniumWebDriverHelper.moveCursorTo(
+        seleniumWebDriverHelper.waitPresence(By.id(TOOLBAR_TITLE_ID)));
+    seleniumWebDriverHelper.getAction().click().perform();
+  }
+
+  public void clickOnInputFieldTag(String tagName) {
+    seleniumWebDriverHelper.waitAndClick(By.xpath(format("//div[text()='%s']", tagName)));
```

Comment:
`"//div[text()='%s']"` looks too simple to reliably prevent confusing with wrong element.